### PR TITLE
fix #383

### LINKED ISSUE #383: Broken translation of Coq inductive declarations into `indt-decl` in the presence of non uniform parameters
Here is an inductive type:
```coq
Inductive t (A : Type) (y : nat) : Type :=
  | K (x : A) {n : nat} : t A n -> t A y.
```
Reading its declaration with `coq.env.indt-decl` gives the following result:
```elpi
parameter A explicit (sort (typ «Test.89»)) c0 \
 inductive t tt 
  (parameter y explicit (pglobal (indt «nat») «») c1 \
	arity (sort (typ «max(Set, Test.89)»))) c1 \
  [constructor K
    (parameter y explicit (pglobal (indt «nat») «») c2 \
      parameter x explicit c1 c3 \
       parameter n maximal (pglobal (indt «nat») «») c4 \
        arity (prod `_` (app [c1, c4]) c5 \ app [c1, c2]))]
```
Please pay attention to the line declaring the `x` parameter in the constructor. It has type `A`, so it should be a binder of type `c0` in the Elpi declaration. Here it's a `c1`, which is not a type but the term `t A`, so the `indt-decl` value generated by Elpi is ill-typed.

### PR DESCRIPTION
**Pull request by gares**

### CODE REVIEW

File: src/coq_elpi_HOAS.ml [hunk 1]
```diff
@@ -3027,16 +3027,17 @@ let hoas_ind2lp ~depth coq_ctx state { params; decl } =
...
     
       let state, arity, gls1 = embed_arity ~depth coq_ctx state (nuparams,typ) in
       let coq_ctx = push_coq_ctx_local depth (Context.Rel.Declaration.LocalAssum(Context.anonR,EConstr.mkProp)) coq_ctx in
       let depth = depth+1 in
       let embed_constructor state { id; arity; typ } =
-        let kctx = arity in
-        let state, karity, gl = embed_arity ~depth coq_ctx state (kctx,reloc kctx typ) in
+        let alen = List.length arity in
+        let kctx = List.mapi (fun i ({ extra; typ } as x) -> { x with typ = reloc (alen - i -1) typ }) arity in
+        let state, karity, gl = embed_arity ~depth coq_ctx state (kctx,reloc (List.length kctx) typ) in
```
Comment (gares) [inline src/coq_elpi_HOAS.ml:3040]:
```suggestion
        let state, karity, gl = embed_arity ~depth coq_ctx state (kctx,reloc alen typ) in
```

**Comment by gares** [comment]
@CohenCyril @proux01 is the CI of 733 skipping odd-order? Is my CI setup broken?
I'd like to test this fix at full, and the failure on odd-order seems unrelated to me.

**Comment by proux01** [comment]
@gares sorry it may have been temporarily broken while I ported the predicate hierarchy in ssralg, now fixed.
To test it here you probably just need an overlay: https://github.com/math-comp/hierarchy-builder/pull/309 (copied from math-comp/math-comp#733 )

**Comment by proux01** [comment]
Sorry, I mixed this for HB, well it can be tested directly in math-comp/math-comp#733 but you would have to edit the .nix/config.nix file since currently only the `coq-master` bundle uses the `coq-master` branch of coq-elpi (other bundles use stable coq-elpi versions as defined in nixpkgs).

**Pull request merged**